### PR TITLE
299102: Tag and Release

### PR DESCRIPTION
**Pull request by rajeshkaremane**
# **What this PR does / why we need it**:
*A brief description of changes being made*
[AB#299102](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/299102)

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=461237&view=results
https://github.com/DEFRA/ffc-demo-web/releases

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [x] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [x] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: docs/AppBuildAndDeploy.md [hunk 1]
```diff
@@ -134,14 +137,16 @@ extends:
     deployFromFeature: ${{ parameters.deployFromFeature }}  #Mandatory: True/False(default)  parameter used to deploy feature branch to dev environment.
     deployConfigOnly: ${{ parameters.deployConfigOnly }}    #Mandatory: True/False(default)  parameter used to deploy app config to various environments.
     privateAgentName: 'DEFRA-ubuntu2204'                    #Optional:  Name of the private build agent. default will use Azure hosted linux agent.
-    packageFeedName: 'artifact-feed'                        #Mandatory: Name of the Azure Devops Artifacts package feed. Used by .Net and NodeJs build.
+    packageFeedName: 'artifact-feed'                        #Optional: Name of the Azure Devops Artifacts package feed. Used by .Net and NodeJs build.
     appBuildConfig: ${{ parameters.appBuildConfig }}        #Mandatory: Object which contains configration used for building the application. Such as appFrameworkType, defaultBranch, frameworkVersion, projectPath, manifestPath, imageRepoName
+      githubreleaseconnection: 'github connection'          #Mandatory: Should be a OAuth or PAT. Used to tag and create github release
```
Comment (DaRoszko) [inline docs/AppBuildAndDeploy.md:142]: Can be defaulted to 'DEFRA' as we'll scaffold projects using that?

Comment (DaRoszko) [reply docs/AppBuildAndDeploy.md:142]: but documentation is right in general

Comment (rajeshkaremane) [reply docs/AppBuildAndDeploy.md:142]: DEFRA connection is not OAuth it is based on "azure pipelines app". Hence had to create new connection.

File: templates/powershell/build/TagRelease.ps1 [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+Write-Host "${functionName} started at $($startTime.ToString('u'))"
+Write-Debug "${functionName}:AppVersion=$AppVersion"
+
+try {
+
+    $exists = git tag -l "$AppVersion"
+    if ($exists) { 
+        Write-Host "Tag already exists"
+    }    
+    git tag $AppVersion --force
```
Comment (ppotturi) [inline templates/powershell/build/TagRelease.ps1:47]: use the PS-Helps Invoke-command for all the git commands?

Comment (rajeshkaremane) [reply templates/powershell/build/TagRelease.ps1]: Fixed

File: docs/AppBuildAndDeploy.md (hunk 1 above)
Comment (DaRoszko) [reply docs/AppBuildAndDeploy.md:142]: @ppotturi  are we sharing the same/correct connection for ADP Portal API?

Comment (ppotturi) [reply docs/AppBuildAndDeploy.md:142]: I didn't get it. Can we have a call?

**Review by kbabigumira — APPROVED** [review]

**Review by ppotturi — APPROVED** [review]

File: docs/AppBuildAndDeploy.md (hunk 1 above)
Comment (rajeshkaremane) [reply docs/AppBuildAndDeploy.md:142]: Added task to the backlog: https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/306312

**Pull request merged**